### PR TITLE
[Feature][Connector-V2]Support Doris Fe Node HA

### PR DESCRIPTION
**Pull request by xiaochen-zhou**
### Purpose of this pull request

The current logic is to randomly select a FE node to connect to, and if it fails, Doris is considered unavailable. We should optimize this by continuously retrying all user-configured FE nodes.

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

exist tests

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by Hisoka-X — COMMENTED** [review]
We should make sure the random and ha be supported at the same time. Random pick the address which can be connected.

**Comment by xiaochen-zhou** [comment]
> We should make sure the random and ha be supported at the same time. Random pick the address which can be connected.

Assume a list of FE nodes: node1, node2, node3, where node1 is unavailable for connection. Our current logic is to randomly select from node1, node2, node3, and if we happen to select node1, we consider this Doris instance unavailable. I think we can optimize this by sequentially selecting from node1, node2, node3, and if node1 is unavailable, then try to establish a connection with node2. This can be considered as HA. I think these two strategies are not contradictory. 
Do you mean to maintain compatibility with the current logic, where if a single node is unavailable, the entire Doris instance is considered unavailable?

**Comment by xiaochen-zhou** [comment]
> We should make sure the random and ha be supported at the same time. Random pick the address which can be connected.

OK, I will modify it according to this idea later

**Comment by xiaochen-zhou** [comment]
> We should make sure the random and ha be supported at the same time. Random pick the address which can be connected.

done.

File: seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/rest/RestService.java [hunk 1]
```diff
@@ -298,16 +283,33 @@ public static List<PartitionDefinition> findPartitions(
...
+                                getUriStr(feNodes.get(i), dorisSourceTable, logger) + QUERY_PLAN);
+                httpPost.setEntity(stringEntity);
+                resStr = send(dorisSourceConfig, httpPost, logger);
+                break;
+            } catch (DorisConnectorException e) {
+                if (i == feNodesNum - 1) {
+                    throw new DorisConnectorException(
+                            DorisConnectorErrorCode.REST_SERVICE_FAILED, e);
+                }
+            }
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/rest/RestService.java]: Can we sure the exception is must cause by FE unavailable?

Comment (xiaochen-zhou) [reply seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/rest/RestService.java]: > Can we sure the exception is must cause by FE unavailable?

I updated the code to catch other potential exceptions.

**Review by Carl-Zhou-CN — APPROVED** [review]
+1

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**